openssl-dynamic/src/main/c/ssl.c

```diff
@@ -721,17 +787,29 @@ TCN_IMPLEMENT_CALL(jint, SSL, initialize)(TCN_STDARGS, jstring engine)
                 if (!ENGINE_set_default(ee, ENGINE_METHOD_ALL))
                     err = APR_ENOTIMPL;
             }
-            /* Free our "structural" reference. */
-            if (ee)
-                ENGINE_free(ee);
+            if (err == APR_SUCCESS) {
+                // This code is based on libcurl:
+                // https://github.com/curl/curl/blob/curl-7_61_0/lib/vtls/openssl.c#L521
+                ui_method = UI_create_method((char *)"netty-tcnative user interface");
+                if (ui_method != NULL) {
+                    UI_method_set_opener(ui_method, UI_method_get_opener(UI_OpenSSL()));
```

Comment:
I'm uncomfortable ignoring error codes from these setters.    Could you make them an assert 1 at least?

Reply:
@carl-mastrangelo sure thing...

Reply:
that said its actually `0` on success https://www.openssl.org/docs/man1.1.0/crypto/UI_method_set_opener.html

Reply:
Well, dang, thats scary.  ui.h says the opposite:  https://github.com/openssl/openssl/blob/master/include/openssl/ui.h#L256

```
   All these functions are expected to return:
        0       on error.
        1       on success.
```

Reply:
I guess this is talking about the "functions" that you use as opener etc itself. not the setter.